src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -487,19 +481,9 @@ private void reconfigureKafkaConsumer(final boolean forceSeek) {
                     }
                 })
                 .map(PartitionStatistics::getBeforeFirst)
-                .collect(groupingBy(NakadiCursor::getEventType));
-    }
-
-    private NakadiCursor getBeforeFirstCursor(final EventTypePartition pk)
-            throws InternalNakadiException, NoSuchEventTypeException, ServiceUnavailableException {
-        final Timeline firstTimelineForET = getContext().getTimelineService()
-                .getActiveTimelinesOrdered(pk.getEventType()).get(0);
-
-        final Optional<PartitionStatistics> stats = getContext().getTimelineService()
-                .getTopicRepository(firstTimelineForET)
-                .loadPartitionStatistics(firstTimelineForET, pk.getPartition());
-
-        return stats.get().getBeforeFirst();
+                .collect(Collectors.toMap(
+                        cursor -> new EventTypePartition(cursor.getEventType(), cursor.getPartition()),
```

Comment:
@rcillo can you please use `cursor::getEventTypePartition` ?

Reply:
But it's not so important actually

Reply:
@antban done bdbf665